SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -736,13 +736,15 @@ FRPCErrorInfo USpatialSender::SendLegacyRPC(UObject* TargetObject, UFunction* Fu
 	Worker_ComponentId ComponentId = SpatialConstants::RPCTypeToWorkerComponentIdLegacy(RPCInfo.Type);
 	if (!NetDriver->StaticComponentView->HasAuthority(EntityId, ComponentId))
 	{
-		bool bShouldDrop = true;
+		ERPCQueueCommand QueueCommand = ERPCQueueCommand::ContinueProcessing;
 		if (AActor* TargetActor = Cast<AActor>(TargetObject))
 		{
-			bShouldDrop = !WillHaveAuthorityOverActor(TargetActor, TargetObjectRef.Entity);
+			if (!WillHaveAuthorityOverActor(TargetActor, TargetObjectRef.Entity))
+			{
+				QueueCommand = ERPCQueueCommand::StopProcessing;
+			}
 		}
-
-		return FRPCErrorInfo{ TargetObject, Function, ERPCResult::NoAuthority, bShouldDrop };
+		return FRPCErrorInfo{ TargetObject, Function, ERPCResult::NoAuthority, QueueCommand };
```

Comment:
Previously, if `WillHaveAuthorityOverActor` was false, it would drop the RPC and keep processing the queue, now it seems to stop processing and instead queues this RPC, which is incorrect. I believe in this case we should also drop the entire queue, as it should be impossible for this worker to gain authority in between processing these RPCs, so it will drop all of them.